src/features/payment/OrderDetailsProvider.tsx

```diff
@@ -27,7 +26,7 @@ export function useOrderDetailsQueryDef(enabled: boolean, instanceId?: string):
 const useOrderDetailsQuery = () => {
   const instanceId = useLaxInstanceId();
   const enabled = useHasPayment();
-  const utils = useQueryWithStaleData<OrderDetails, HttpClientError>(useOrderDetailsQueryDef(enabled, instanceId));
+  const utils = useQuery<OrderDetails, HttpClientError>(useOrderDetailsQueryDef(enabled, instanceId));
```

Comment:
I don't know exactly why this used `useQueryWithStaleData`, but I would assume it has to do with invalidating the query after initial loading, and not wanting the UI to flash a loading indicator? Have you tested an app with payment before/after this change?